titiler/pgstac/factory.py

```diff
@@ -699,6 +718,166 @@ def info_search(request: Request, searchid=Depends(self.path_dependency)):
+            ),
+            sortby: Optional[str] = Query(
+                None,
+                description="Sort the response items by a property (ascending (default) or descending).",
+            ),
+        ):
+            """List a Search query."""
+            # Default filter to only return `metadata->type == 'mosaic'`
+            mosaic_filter = sql.SQL("metadata->>'type' = 'mosaic'")
+
+            # additional metadata property filter passed in query-parameters
+            # <propname>=val - filter for a metadata property. Multiple property filters are ANDed together.
+            qs_key_to_remove = ["limit", "offset", "sortby"]
+            additional_filter = [
+                sql.SQL("metadata->>{key} = {value}").format(
+                    key=sql.Literal(key), value=sql.Literal(value)
+                )
+                for (key, value) in request.query_params.items()
+                if key.lower() not in qs_key_to_remove
+            ]
+            filters = [
+                sql.SQL("WHERE"),
+                sql.SQL("AND ").join([mosaic_filter, *additional_filter]),
+            ]
+
+            def parse_sort_by(sortby: str) -> Generator[sql.Composable, None, None]:
+                """Parse SortBy expression."""
+                for s in sortby.split(","):
+                    parts = re.match(
+                        "^(?P<dir>[+-]?)(?P<prop>.*)$", s.lstrip()
+                    ).groupdict()  # type:ignore
+
+                    prop = parts["prop"]
+                    if parts["prop"] in ["lastused", "usecount"]:
+                        prop = sql.Identifier(prop)
+                    else:
+                        prop = sql.SQL("metadata->>{}").format(sql.Literal(prop))
+
+                    if parts["dir"] == "-":
+                        order = sql.SQL("{} DESC").format(prop)
+                    else:
+                        order = sql.SQL("{} ASC").format(prop)
+
+                    yield order
+
+            # sortby=[+|-]PROP - sort the response items by a property (ascending (default) or descending).
+            order_by = []
+            if sortby:
+                sort_expr = list(parse_sort_by(sortby))
+                if sort_expr:
+                    order_by = [
+                        sql.SQL("ORDER BY"),
+                        sql.SQL(", ").join(sort_expr),
+                    ]
+
+            with request.app.state.dbpool.connection() as conn:
+                with conn.cursor() as cursor:
+                    # Get Total Number of searches rows
+                    query = [
+                        sql.SQL("SELECT count(*) FROM searches"),
+                        *filters,
+                    ]
+                    cursor.execute(sql.SQL(" ").join(query))
+                    nb_items = int(cursor.fetchone()[0])
+
+                    # Get rows
+                    cursor.row_factory = class_row(model.Search)
+                    query = [
```

Comment:
why are you doing the query as an array here?

Reply:
why not? I'm just creating a `List` of operation and then use `sql.SQL(" ").join(query)`